airflow/models/taskinstance.py

```diff
@@ -532,6 +569,41 @@ def __init__(
         # can be changed when calling 'run'
         self.test_mode = False
 
+    @staticmethod
+    def from_task(
+        task: Operator,
+        execution_date: datetime | None = None,
+        run_id: str | None = None,
+        state: str | None = None,
+        map_index: int = -1,
+    ):
+        """
+        Create TaskInstance from task operator.
+
+        :param task: task's Operator object.
+        :param execution_date: Optional execution time of the task.
+        :param run_id: Optional DAG run ID for the task.
+        :param state: Optional state of the task.
+        :param map_index: Optional map index. Defaults to -1 (non-mapped task).
+        """
+        return TaskInstance(task, execution_date, run_id, state, map_index)
+
+    @staticmethod
+    def deserialize(ti_dict: dict[str, Any], version: int) -> TaskInstance:
+        """Deserialize TaskInstance from dictionary."""
+        if version > TaskInstance.__version__:
+            raise TypeError("version too big, dont know hot to deserialize")
```

Comment:
Mmm you have even copied my typo ;-)

Reply:
I suggest clarifying it a bit, by being more explicit (what version did you get and what do you support as a max)